### ISSUE TITLE
Implement no-unused-vars

### ISSUE THREAD

**Issue by andersekdahl** [yannickcr/eslint-plugin-react#61]
Feature request for shadowing ESLints `no-unused-vars` rule. Currently if you have:

```
<button onClick={e => this.close()}>Close</button>
```

You're not warned that you have an unused variable (`e`).

**Comment by yannickcr** [comment]
This case is handled by ESLint so it should already warn you:

.eslintrc
```json
{
    "env": {
        "es6": true
    },
    "ecmaFeatures": {
        "jsx": true
    },
    "rules": {
        "no-unused-vars": 2
    }
}
```
test.jsx
```js
<button onClick={e => this.close()}>Close</button>
```

```bash
$ eslint test.jsx --reset

test.jsx
  1:17  error  e is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

Tested with ESLint 0.19.0

**Comment by andersekdahl** [comment]
Sorry for this, I noticed it in a project which allows unused arguments in .eslintrc. Can confirm that this works as expected.
